mbedtls/src/cipher/raw/serde.rs

```diff
@@ -341,20 +341,20 @@ unsafe impl BytesSerde for gcm_context {}
 
 // If the C API changes, the serde implementation needs to be reviewed for correctness.
 
-unsafe fn _check_cipher_context_t_size(ctx: cipher_context_t) -> [u8; 96] {
+unsafe fn _check_cipher_context_t_size(ctx: cipher_context_t) -> [u8; size_of::<cipher_context_t>()] {
```

Comment:
I think we need to use constants here, as compilation of these functions alerts us of changes in these objects.

E.g. if for some reason `cipher_context_t` changes size, then this will result in a compilation failure. However using `cipher_context_t` against its own size will always work. 

Does this make sense?

Reply:
Ah yes, I see, this definitely makes sense. But the value of that constant must be platform-dependent. So, it should roughly be `x + y * size_of::<usize>()` I guess?

Reply:
I don't think so, all these structs fields are only dependent on crypto (see e.g. https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/cipher.h#L319 for `cipher_context_t`). E.g. any platform should serialize a `cipher_context_t` in exactly 96 bytes.

(Orthogonally, I do think that these constants should be declared separately e.g. `const CIPHER_CONTEXT_T_SIZE: usize = 96;`)

Reply:
Okay, I haven't checked this specific check in detail but I have changed it because some of the checks failed when building for `thumbv7em-none-eabihf` so I just assumed now this would be due to different `usize` sizes. I will take a look at this again soon.

Reply:
Oh, different _usize_ sizes? I thought `usize` is always 8 bits.
You are right, we should adjust by target (and perhaps add some sanity check test that checks that serialized objects are really exactly 96, 288, 128, etc bytes, if possible)

Reply:
`usize` is _the_ type that comes to my mind when talking about different sizes in memory because it is
> The pointer-sized unsigned integer type.
> 
> The size of this primitive is how many bytes it takes to reference any location in memory. For example, on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.

See https://doc.rust-lang.org/std/primitive.usize.html

Reply:
Thanks, so let us adjust those constants according to the target.

Reply:
I had a look at the underlying C structs and calculated the expected sizes. On a 64 bit machine, this works. I still have to test this when building for a 32 bit target.

Reply:
With the latest commit, this also works for 32 bit targets 👍